### PR TITLE
[cu-2w8zbnh] Make RewardPoolId same as the staked assetId

### PR DESCRIPTION
**Pull request by vimukthi-git**
Refer issue

### CODE REVIEW

**Comment by ComposableFiB0t** [comment]
Task linked: [CU-2w8zbnh Add PICA and PBLO stake pools as part of genesis configuration of the staking rewards pallet and not create in Pablo pallet at pool creation time](https://app.clickup.com/t/2w8zbnh)

**Review by benluelo — APPROVED** [review]

File: frame/staking-rewards/src/test/test_reward_accumulation_hook.rs [hunk 1]
```diff
@@ -505,7 +505,7 @@ fn test_accumulate_rewards_hook() {
 			]);
 
 			check_events([crate::Event::<Test>::MaxRewardsAccumulated {
-				pool_id: bobs_pool_id,
+				pool_id: C::ID,
```
Comment (benluelo) [inline frame/staking-rewards/src/test/test_reward_accumulation_hook.rs:508]: was this change required? (curious why there were changes in `check_events` and not in `check_rewards`)

Comment (vimukthi-git) [reply frame/staking-rewards/src/test/test_reward_accumulation_hook.rs:508]: It's because we no longer require an explicit `pool_id` to verify rewards, staked asset ID(`pool_asset_id`) is already the pool id. Made a change to get rid of it in the test anyway

**Review by benluelo — APPROVED** [review]

**Review by andor0 — APPROVED** [review]

**Review by hussein-aitlahcen — APPROVED** [review]
